elastic/eui
# [EuiSuperDatePicker] Fix broken EuiDatePickerRange styling; Add `isDisabled` object config

## Summary

This PR accomplishes two separate tasks:

1. (first two commits) I added an extra opinionated `<span>` to `EuiDatePickerRange` in https://github.com/elastic/eui/pull/6795/commits/fa7bc446e51595e876308ee503b08aee7505675d, totally forgetting that `EuiSuperDatePicker` dogfoods `EuiDatePickerRange`, and consequently [broke range styling](https://eui.elastic.co/v81.1.0/#/templates/super-date-picker) on our latest release.
    <img width="1384" alt="before/after screenshot" src="https://github.com/elastic/eui/assets/549407/36e887db-c9b6-43da-a7ba-85f61691f701">
      - This PR fixes that and additionally cleans up the DOM around EuiSuperDatePicker by removing `.euiSuperDatePicker__flexWrapper` (and flex children wrappers), and instead placing the `.euiSuperDatePicker` CSS class directly on the topmost wrapper:
        <img width="700" alt="screenshot of HTML" src="https://github.com/elastic/eui/assets/549407/f3d4e43c-5617-4caa-9f23-29a48252a848">


2. (last two commits) closes https://github.com/elastic/eui/issues/6814
    - This PR adds a new `isDisabled={{ display }}` object configuration which allows consumers to customized the displayed disabled text (e.g., if they want to render text saying that no times are currently being filtered).

<img width="370" alt="screenshot" src="https://github.com/elastic/eui/assets/549407/21199c33-e7e0-4eaa-8660-0a2cd8d8e707">

## QA

- Go to http://localhost:8030/#/templates/super-date-picker
  - [ ] Confirm the first example is grayed out/disabled correctly with custom disabled text
- Go to http://localhost:8030/#/templates/super-date-picker#update-button
  - Click the input to toggle the date picker range
  - [ ] Confirm that the revealed date picker range works as it did in [v81.0.0](https://eui.elastic.co/v81.0.0/#/templates/super-date-picker) and is not broken width-wise like in [v81.1.0](https://eui.elastic.co/v81.1.0/#/templates/super-date-picker)

### General checklist


~- [ ] Checked for **breaking changes** and labeled appropriately~ DOM/CSS changes are semi-breaking, but not part of the promised API, and I'll update various Kibana references accordingly
~- [ ] Added **[documentation](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting)**~ I'm skipping adding a new/specific documentation section for this behavior. TBH, I consider it kinda edge case and for a very specific Kibana usage, and I'd rather document/QA it only in Storybook rather than encourage usage.
